### PR TITLE
Fixed issue #10170: Yii::powered() does not use i18n

### PR DESCRIPTION
**Pull request by SamMousa**

### CODE REVIEW

File: framework/BaseYii.php [hunk 1]
```diff
@@ -464,7 +464,10 @@ public static function endProfile($token, $category = 'application')
      */
     public static function powered()
     {
-        return 'Powered by <a href="http://www.yiiframework.com/" rel="external">Yii Framework</a>';
+        return static::t('yii', 'Powered by {beginLink}Yii Framework{endLink}', [
```
Comment (samdark) [inline framework/BaseYii.php]: I'd do it as:

```php
return static::t('yii', 'Powered by {yii}', [
    'yii' => '<a href="http://www.yiiframework.com/" rel="external">' . static::t('yii', Yii Framework') . '</a>'
]);
```

Comment (cebe) [inline framework/BaseYii.php]: does the term "Yii Framework" really need a translation?

**Comment by cebe** [comment]
Event though calling `static::t()` seems obvious, the message extractor will not find it.

File: framework/BaseYii.php (hunk 1 above)
Comment (samdark) [inline framework/BaseYii.php]: In Russian, yes. "Framework" part of it.

**Comment by SamMousa** [comment]
Can I add it in comments? Or does the message extractor skip comments? 
IE:

       // Yii::t('yii', 'Same string here')

It would mean minor (but really local) duplication.

**Comment by SilverFire** [comment]
Are we required to use `static::t()`? Maybe we can just leave it `Yii::t()`?

**Comment by SamMousa** [comment]
Hmm, good point, if we use Yii::t everywhere else in the core subclassing
Yii and overriding t won't work anyway ...

**Comment by cebe** [comment]
the only difference between Yii::t and static::t would be that the subclass must be named Yii if you want to change behavior of Yii::t. This is the case in 99,99% and if you then really want to use Yii::powered, you can extend it too. So I think using Yii::t is the way to go.

**Comment by SamMousa** [comment]
Note that this adds a dependency in the wrong direction.

(Although I agree this is the most viable option)

**Comment by SilverFire** [comment]
Merged with reword in Changelog https://github.com/yiisoft/yii2/commit/6d7627c2a51690175875fb8f21d444a3fb656ba6

**Comment by SilverFire** [comment]
Thank you!

**Pull request closed without merging**